pkg-manager/real-hoist/src/index.ts

```diff
@@ -10,6 +10,65 @@ export type HoistingLimits = Map<string, Set<string>>
 
 export type { HoisterResult }
 
+/**
+ * In the (default) "none" mode, dependencies are hoisted as far as possible.
+ * For example, given workspace package "A" which depends on package "B" which
+ * depends on package "C", the layout on disk will be:
+ *
+ * - /packages/A
+ * - /node_modules/B
+ * - /node_modules/C
+ *
+ * In the "workspaces" mode, dependencies are hoisted only as far as each
+ * workspace package. Given the same example, the layout would be:
+ *
+ * - /packages/A
+ * - /packages/A/node_modules/B
+ * - /packages/A/node_modules/C
+ *
+ * In the "dependencies" mode, dependencies are only hoisted up to the direct
+ * dependencies of each workspace package. Given the same example:
+ *
+ * - /packages/A
+ * - /packages/A/node_modules/B
+ * - /packages/A/node_modules/B/node_modules/C
+ */
+export type HoistingLimitMode = 'none' | 'workspaces' | 'dependencies'
+
+export function getHoistingLimits (lockfile: Lockfile, mode: HoistingLimitMode | undefined): HoistingLimits | undefined {
```

Comment:
there is no need to make this function public. Just call it from the hoist function instead of passing `hoistingLimits` to the `hoist` function.

Reply:
I guess I can shove it down the stack; the main thing I was trying to reconcile is the fact that this API takes in hoisting limits and so I need to respect them. But, I guess I can just do this somewhere else.

Reply:
There's no need to take in hoisting limits. It is not used.

Reply:
I'm not following; are you saying that #4300 can in effect be reverted? That is the API I'm referring to, and it's exposed all the way up to `headlessInstall`.

Reply:
yes, I needed it for Bit but then I came up with a different solution. So, it is not needed.

Reply:
I'm happy to revert that API and make this logic private, if you're confident that it's safe to do so! I'm not totally sure what API guarantees you make, so I went with the conservative option.

Reply:
It is OK. It will be a major version bump.